### PR TITLE
Fixes various sonar cloud issues

### PR DESCRIPTION
**Pull request by matthias-ronge**
- Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed.
- Remove usage of generic wildcard type.
- Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed.
- Refactor this method to reduce its Cognitive Complexity from 27 to the 15 allowed.
- Remove usage of generic wildcard type.
- 8 duplicated blocks of code must be removed.
- 9 duplicated blocks of code must be removed.
- Move the contents of this initializer to a standard constructor or to field initializers.
- 3 duplicated blocks of code must be removed.
- 5 duplicated blocks of code must be removed.
- 3 duplicated blocks of code must be removed.
- duplicated blocks of code must be removed.
- 4 duplicated blocks of code must be removed.
- 1 duplicated blocks of code must be removed.
- 1 duplicated blocks of code must be removed.
- Rename "ruleset" which hides the field declared at line 48.
- 2 duplicated blocks of code must be removed.
- Refactor this method to reduce its Cognitive Complexity from 32 to the 15 allowed.
- Remove usage of generic wildcard type.
- 1 duplicated blocks of code must be removed.
- 1 duplicated blocks of code must be removed.
- Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed.
- 1 duplicated blocks of code must be removed.
- 1 duplicated blocks of code must be removed.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/helper/metadata/MetadataHelper.java [hunk 1]
```diff
@@ -46,6 +47,22 @@
     private static final int PAGENUMBER_FIRST = 0;
     private static final int PAGENUMBER_LAST = 1;
     private LegacyPrefsHelper prefs;
+    private final Comparator<? super LegacyReferenceHelper> referencesSorter = (firstObject, secondObject) -> {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/helper/metadata/MetadataHelper.java]: Why not in separated method?

**Comment by Kathrin-Huber** [comment]
please resolve conflicts

File: Kitodo/src/main/java/org/kitodo/production/helper/metadata/legacytypeimplementations/LegacyDocStructHelperInterface.java [hunk 2]
```diff
@@ -351,5 +400,39 @@ default String getAnchorClass() {
...
+     * This method generates a comprehensible log message in case something was
+     * overlooked and one of the unimplemented methods should ever be called in
+     * operation. The name was chosen deliberately short in order to keep the
+     * calling code clear.
+     * 
+     * @param exception
+     *            created {@code UnsupportedOperationException}
+     * @return the exception
+     */
+    static RuntimeException andLog(UnsupportedOperationException exception) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/helper/metadata/legacytypeimplementations/LegacyDocStructHelperInterface.java]: why don't you return "UnsupportedOperationException?

File: Kitodo/src/main/java/org/kitodo/production/helper/tasks/ExportNewspaperBatchTask.java [hunk 3]
```diff
@@ -562,6 +566,14 @@ private static Integer positionByRank(List<LegacyDocStructHelperInterface> sibli
         return result;
     }
 
+    private static int[] _f(List<LegacyMetadataHelper> allMetadata) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/helper/tasks/ExportNewspaperBatchTask.java]: you are not using this method!?
Why does it have such a strange name?

File: Kitodo/src/main/java/org/kitodo/production/helper/tasks/ExportNewspaperBatchTask.java [hunk 4]
```diff
@@ -46,6 +46,9 @@
 
 public class ExportNewspaperBatchTask extends EmptyTask {
     private static final Logger logger = LogManager.getLogger(ExportNewspaperBatchTask.class);
+
+    private static final String CANNOT_DETERMINE_POSITION = "Cannot determine position to place {} correctly because the "
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/helper/tasks/ExportNewspaperBatchTask.java]: this should go to errors-en.properties

**Pull request merged**